client/src/settings/mdn-worker.tsx

```diff
@@ -73,7 +78,10 @@ export class MDNWorker {
   }
 
   registerMessageHandler() {
-    navigator.serviceWorker.addEventListener("message", this.messageHandler);
+    const self = this;
```

Comment:
@LeoMcA please also verify that this is needed. Just look at `window.mdnWorker.mutationCounter` in the console after adding / deleting a collection item.

Reply:
ah yes, it's half needed - what threw me last time was the `const self = this`, which I believe is still unnecessary because of how arrow functions don't bind `this` - but the arrow function *is* still necessary because otherwise `this` within `messageHandler` gets bound to `navigator.serviceWorker`. So, I think all that's necessary here is (and seems to work in testing):

```js
navigator.serviceWorker.addEventListener("message", (e) => this.messageHandler(e));
```